pkg/agent/controller/networkpolicy/cache.go

```diff
@@ -152,26 +153,33 @@ type ruleCache struct {
 	podUpdates <-chan v1beta1.PodReference
 }
 
-func (c *ruleCache) getNetworkPolicies(namespace string) []v1beta1.NetworkPolicy {
-	ret := []v1beta1.NetworkPolicy{}
+func (c *ruleCache) getNetworkPolicies(npFilter querier.NetworkPolicyQueryFilter) []v1beta1.NetworkPolicy {
```

Comment:
It doesn't seem the argument and `NetworkPolicyReference` in policyMap and the following methods need to be struct, which could introduce some copy overhead in each iteration I guess.

Reply:
Do you mean we can use several arguments instead of using `NetworkPolicyQueryFilter` and `NetworkPolicyReference` in the policyMap? 
The reason that I designed `NetworkPolicyQueryFilter` struct is that I want to make it extensible. In the future, if we need more arguments to retrieve a policy we can just change this struct and that compare function which is `networkPolicyMatchFilter()`.
About the `NetworkPolicyReference` in the policyMap, I think since `NetworkPolicyReference` contains less enough attributes to locate a policy, I change the value of policyMap from `NamespacedName` to `NetworkPolicyReference`.
Emm... I'm not sure if this will cause an additional copy overhead issue. I'm glad to hear your suggestions.

Reply:
No, I meant using pointer instead of value of the struct as the arguments and the value of map to save the copy overhead in multiple places.

Reply:
Oh got you. Make sense.